app/domain/people_skills_filter.rb

```diff
@@ -25,7 +26,33 @@ def filter_by_rated
     entries
   end
 
-  def filter_by_level(entries)
-    level.nil? ? entries : entries.where('level >= ?', level)
+  def filter_by_level(entries, skills)
+    return entries if level.nil?
+    skill_ids = skills.to_s.split(',')
+    if skill_ids.length == @level.length
+      return get_filter_results(entries, skill_ids)
+    else
+      raise ArgumentError, 'Amount of Skill_ids and levels provided do not match'
+    end
+  end
+
+  def get_filter_results(entries, skill_ids)
```

Comment:
rename to `level_filter_results`. Also, you don't need to pass the skill_ids here since it's an instance variable
`get_filter_results` is very general